Don't refer to XCTest from Sources

This seems wrong?

```
#0 34.49 [171/241] Emitting module OTel
#0 34.49 /usr/src/libs/___/.build/checkouts/swift-otel/Sources/OTel/Helpers/HookedFatalError.swift:15:8: error: no such module 'XCTest'
#0 34.49 import XCTest
```

Got a report like this from RHEL user

--- a/Sources/OTel/Helpers/HookedFatalError.swift
+++ b/Sources/OTel/Helpers/HookedFatalError.swift
@@ -11,8 +11,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-import struct NIOConcurrencyHelpers.NIOLockedValueBox
 import class Foundation.Thread
+import struct NIOConcurrencyHelpers.NIOLockedValueBox
 
 func fatalError(_ message: @autoclosure () -> String = String(), file: StaticString = #file, line: UInt = #line) -> Never {
     print("hooking fatal error")

--- a/Sources/OTel/Helpers/HookedFatalError.swift
+++ b/Sources/OTel/Helpers/HookedFatalError.swift
@@ -12,7 +12,7 @@
 //===----------------------------------------------------------------------===//
 
 import struct NIOConcurrencyHelpers.NIOLockedValueBox
-import XCTest
+import class Foundation.Thread
 
 func fatalError(_ message: @autoclosure () -> String = String(), file: StaticString = #file, line: UInt = #line) -> Never {
     print("hooking fatal error")

